verticaltabs.jsm

```diff
@@ -596,25 +601,35 @@ VerticalTabs.prototype = {
       aTab.setAttribute('crop', 'end');
     }
     let tab_meta_image = document.getAnonymousElementByAttribute(aTab, 'anonid', 'tab-meta-image');
-    tab_meta_image.style.backgroundImage = `url(moz-page-thumb://thumbnail/?url=${encodeURIComponent(aTab.linkedBrowser.currentURI.spec)}), url(resource://tabcenter/skin/blank.png)`;
+    tab_meta_image.style.backgroundImage = `url(moz-page-thumb://thumbnail/?url=${encodeURIComponent(getUri(aTab).spec)}), url(resource://tabcenter/skin/blank.png)`;
+
+    let uri = getUri(aTab);
+    let label = uri.spec;
+    if (uri.scheme.startsWith('http')) {
+      label = uri.host;
+    }
+    // URI can be shown immediately
+    document.getAnonymousElementByAttribute(aTab, 'anonid', 'address-label').value = label;
 
     aTab.addEventListener('load', () => {
-      let url = aTab.linkedBrowser.currentURI.spec;
+      let url = getUri(aTab).spec;
       if (aTab.VTLastUrl !== url) {
         if (url === 'about:newtab' || url === 'about:blank') {
           tab_meta_image.style.backgroundImage = 'url("resource://tabcenter/skin/newtab.png")';
         } else {
           PageThumbs.captureAndStoreIfStale(aTab.linkedBrowser, function (success) {
-            if (aTab.linkedBrowser.currentURI.spec === url) {
+            if (getUri(aTab).spec === url) {
               tab_meta_image.style.backgroundImage = `url('moz-page-thumb://thumbnail/?url=${encodeURIComponent(url)}'), url(resource://tabcenter/skin/blank.png)`;
             }
           });
         }
         aTab.VTLastUrl = url;
-        let label = aTab.linkedBrowser.currentURI.spec;
-        if (aTab.linkedBrowser.currentURI.scheme.startsWith('http')) {
-          label = aTab.linkedBrowser.currentURI.host;
+        let uri = getUri(aTab);
+        let label = uri.spec;
+        if (uri.scheme.startsWith('http')) {
+          label = uri.host;
         }
+        // URI might have changed due to redirects
         document.getAnonymousElementByAttribute(aTab, 'anonid', 'address-label').value = label;
```

Comment:
I think, since we call this twice, we should extract this code into a `setAddressLabel(aTab)` function…
(Also, that would give us a single place to change the Activity Stream's url when we do that.)